### PR TITLE
feat: Added e2e test paid booking should be able to be rescheduled

### LINKED ISSUE #9368: [CAL-1882] Make E2E test: Paid booking should be able to be rescheduled
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1882](https://linear.app/calcom/issue/CAL-1882/make-e2e-test-paid-booking-should-be-able-to-be-rescheduled)</sub>

### PR DESCRIPTION
**Pull request by JaideepGuntupalli**
## What does this PR do?

Added a new `paid booking should be able to be rescheduled` test.

Fixes #9368
/claim #9368

## Type of change
  - Chore (refactoring code, technical debt, workflow improvements)
  - New feature (non-breaking change which adds functionality)

## How should this be tested?

  - Are there environment variables that should be set?
     - Yes, Stripe keys

## Mandatory Tasks

  - [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/playwright/integrations-stripe.e2e.ts [hunk 1]
```diff
@@ -69,9 +69,47 @@ test.describe("Stripe integration", () => {
...
+  test("Paid booking should be able to be rescheduled", async ({ page, users }) => {
+    const user = await users.create();
+    const eventType = user.eventTypes.find((e) => e.slug === "paid") as Prisma.EventType;
+    await user.apiLogin();
+    await page.goto("/apps/installed");
+
+    await user.getPaymentCredential();
+    await user.setupEventWithPrice(eventType);
+    await user.bookAndPaidEvent(eventType);
+
```
Comment (zodman) [inline apps/web/playwright/integrations-stripe.e2e.ts:82]: here is missing the part of check if eventType was paid with a stripe transaction.

File: apps/web/playwright/integrations-stripe.e2e.ts [hunk 2]
```diff
@@ -69,9 +69,47 @@ test.describe("Stripe integration", () => {
...
+    await Promise.all([page.waitForURL("/booking/*"), page.click('[data-testid="reschedule-link"]')]);
+
+    // wait for reschedule page to load
+    await page.waitForSelector('[data-testid="incrementMonth"]');
+
+    await page.click('[data-testid="incrementMonth"]');
+
+    // eslint-disable-next-line playwright/no-wait-for-timeout
+    await page.waitForTimeout(1000);
```
Comment (zodman) [inline apps/web/playwright/integrations-stripe.e2e.ts:91]: This makes the test slower Don't do that!

File: apps/web/playwright/integrations-stripe.e2e.ts [hunk 3]
```diff
@@ -69,9 +69,47 @@ test.describe("Stripe integration", () => {
...
+
+    await page.locator('[data-testid="day"][data-disabled="false"]').nth(1).click();
+    await page.locator('[data-testid="time"][data-disabled="false"]').nth(0).click();
+
+    await page.click('[data-testid="confirm-reschedule-button"]');
+
+    await Promise.all([page.waitForURL("/payment/*"), page.press('[name="email"]', "Enter")]);
+
+    const stripeFrame = page.frameLocator("iframe").first();
```
Comment (zodman) [inline apps/web/playwright/integrations-stripe.e2e.ts:100]: You are duplicating code, please encapsulate these operations in the `users` object  for being  reused.

File: apps/web/playwright/integrations-stripe.e2e.ts [hunk 4]
```diff
@@ -69,9 +69,47 @@ test.describe("Stripe integration", () => {
...
+    await page.goto("/apps/installed");
+
+    await user.getPaymentCredential();
+    await user.setupEventWithPrice(eventType);
+    await user.bookAndPaidEvent(eventType);
+
+    await Promise.all([page.waitForURL("/booking/*"), page.click('[data-testid="reschedule-link"]')]);
+
+    // wait for reschedule page to load
+    await page.waitForSelector('[data-testid="incrementMonth"]');
```
Comment (zodman) [inline apps/web/playwright/integrations-stripe.e2e.ts:86]: this operation it is really needed ? the `page.click` have to wait the selector.
https://playwright.dev/docs/api/class-page#page-click

**Review by zodman — CHANGES_REQUESTED** [review]

File: apps/web/playwright/integrations-stripe.e2e.ts (hunk 4 above)
Comment (JaideepGuntupalli) [reply apps/web/playwright/integrations-stripe.e2e.ts]: fixed this! pls look at new commits

File: apps/web/playwright/integrations-stripe.e2e.ts (hunk 3 above)
Comment (JaideepGuntupalli) [reply apps/web/playwright/integrations-stripe.e2e.ts]: fixed this! pls look at new commits

File: apps/web/playwright/integrations-stripe.e2e.ts (hunk 2 above)
Comment (JaideepGuntupalli) [reply apps/web/playwright/integrations-stripe.e2e.ts]: fixed this! pls look at new commits

File: apps/web/playwright/integrations-stripe.e2e.ts (hunk 1 above)
Comment (JaideepGuntupalli) [reply apps/web/playwright/integrations-stripe.e2e.ts:82]: I am first booking a paid event and then trying to reschedule. So i just followed first steps from Can book a paid booking test in the same file.

**Review by emrysal — APPROVED** [review]
Looks like a good test to me @JaideepGuntupalli - thanks for the first review pass @zodman very good observations that would have indeed blocked this from going in. It's all good now 👏

**Pull request merged**